packages/compass-collection/src/components/collection-tab-provider.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useContext } from 'react';
+import type { CollectionTabPluginMetadata } from '../modules/collection-tab';
+
+export interface CollectionTabPlugin {
+  name: string;
+  component: React.ComponentType<CollectionTabPluginMetadata>;
+}
+
+const CollectionTabsContext = React.createContext<CollectionTabPlugin[]>([]);
+
+export const CollectionTabsProvider: React.FunctionComponent<{
+  tabs: CollectionTabPlugin[];
+}> = ({ tabs, children }) => {
```

Comment:
@gribnoysup I saw you used `useRef()` for the tabs for the instance tabs provider, was there a specific reason for that? I wouldn’t assume that `tabs` change over the lifetime of the component, but if it _did_ change, it would probably do so for a reason, so wouldn’t we want to also change the value in the provider itself?

Reply:
I'm assuming we always want the provider to work as initial configuration for tabs and using a ref removes the need to use memo when rendering a provider. We can change that if you feel like it and to be consistent probably better to do it on other places too them, just need to adjust other places where we already have providers like that to account for that

Reply:
> we’re not changing tabs lists at runtime

That's kinda my point, if we don't ref this value we're now need to keep in mid that passing values there in runtime will actually change it and will rerender children using this context unneccessarily if we're not careful, making you think more about it, not less, which I'm trying to avoid. As I said though, feel free to update this, just also need to move other tab props outside of the render / memo them so that we don't actually update the value in runtime